### DISCUSSION TITLE
PC transversal particle info trace back

### DISCUSSION THREAD

**Discussion by soothing35** [zhonge/cryodrgn#395] [Q&A]
Hi, Developers . Such honor of me to get the chance to use cryodrgn. It is really a powerful tool for EM analysis.
I have successfully generated continuous volume maps using your software. However, I am now seeking further guidance on how to trace back and obtain information about corresponding particles, particularly when employing PC trajectories or graph traversal methods to generate long trajectories. I could get the specific k-means class particle .pkl files. However, in graph traversal methods, I could not find a way to trace back the particle info from the continues map.
In pc1 or pc2 folds of analysis, I have 10 volume mrc files and one z_values.txt. Maybe the z_values.txt file could trace the particle infomation. What  could I do to get this kind of tracing issue?
Hope for your suggestion!

**Comment by michal-g** [comment]
Hi, to clarify, are you looking for how to recover the indices of any particular set of particles (not necessarily kmeans-clustering centroids) to use as input for graph traversal, using the latent space embedding as a reference? Or are you looking for the indices of particles selected as part of e.g. a graph traversal path?

In the former case, the easiest way to accomplish this is without loading the latent space embedding data into a Python session and recovering the indices manually is to use the `cryodrgn filter` command, which interactively displays the indices of particles that are being hovered over by the mouse. For example, if `006_train-vae` is the name of your cryoDRGN output folder, and you have already run `cryodrgn analyze` using epoch 10, you can run `cryodrgn filter 006_train-vae --epoch 10` to get something like this:

![Screenshot 2024-09-03 at 7 47 04 PM](https://github.com/user-attachments/assets/7b41e6a7-6e3f-4dbf-8d3e-7ebc1922869f)
Note the annotation text in the lower-left corner of the screen, as well as the menu options to select other covariate axes to plot the particle points on!

In the latter case, using the `--outind` argument with `graph_traversal` will also output the indices of the particles included in the constructed path to a text file.

Hope this helps!
